pkg/sensors/tracing/killer.go

```diff
@@ -125,7 +125,10 @@ func selectOverrideMethod(specOpts *specOptions) (OverrideMethod, error) {
 			return OverrideMethodInvalid, fmt.Errorf("option override return set, but it is not supported")
 		}
 	case OverrideMethodFmodRet:
-		if !bpf.HasModifyReturnSyscall() {
+		if hasSyscall && !bpf.HasModifyReturnSyscall() {
+			return OverrideMethodInvalid, fmt.Errorf("option fmod_ret on syscall set, but it is not supported")
+		}
+		if hasNormal && !bpf.HasModifyReturn() {
 			return OverrideMethodInvalid, fmt.Errorf("option fmod_ret set, but it is not supported")
```

Comment:
I don't think the `hasNormal` check is needed.

AFIACT, if `HadModifyReturnSyscall()` is true , `HasModifyReturn()` is (always)  true.
So I think the following check:

```
if !bpf.HasModifyReturn() || (hasSyscall && !bpf.HasModifyReturnSyscall()) {
     return err
}
``` 

catches all the cases we care about, without the extra argument.

Reply:
right, I guess I wanted it to be more explicit.. I'll change that